Library/Homebrew/language/node.rb

```diff
@@ -22,8 +22,19 @@ def self.std_npm_install_args(libexec)
       # tell npm to not install .brew_home by adding it to the .npmignore file
       # (or creating a new one if no .npmignore file already exists)
       open(".npmignore", "a") { |f| f.write("\n.brew_home\n") }
+      # npm 5.0.0 broke the assumption Homebrew made that npm would always
+      # create and install real directories when passed a file path. We can
+      # restore the old behaviour by packing the path first with "npm pack"
+      # which also helpfully spits out the filename it just created allowing
+      # us to pass that as the final argument.
+
       # npm install args for global style module format installed into libexec
-      ["--verbose", "--global", "--prefix=#{libexec}", "."]
+      %W[
+        --verbose
+        --global
+        --prefix=#{libexec}
+        #{Dir.pwd}/#{Utils.popen_read("npm pack").chomp}
```

Comment:
The package name actually went first in the example I was recalling, where this did work:
```
      system "npm", "install", "yarn", *Language::Node.std_npm_install_args(buildpath/"vendor/yarn")
      system buildpath/"vendor/yarn/bin/yarn", "install"
```

>Also it looks to contradict against the homebrew philosopy that foo should be vendored as a resource too in such a use case.

We have an exception for things installed via other trusted package managers (e.g. venv.pip_install, cabal_install, etc.)